packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -2436,4 +2442,26 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
       // },
     },
   };
+
+  private callWithGasMargin = async (
+    contract: ethers.Contract,
+    method: string,
+    args: any[],
+    overrides: ethers.PayableOverrides & { from?: string } = {},
+    percentageIncrease: number = GAS_LIMIT_INCREASE_PERCENTAGE
+  ) => {
+    if (!overrides.gasLimit) {
+      const txData = await contract.populateTransaction[method](
+        ...args,
+        overrides
+      );
+      const gasEstimation = await contract.provider.estimateGas(txData);
+      const adjustedGasLimit = gasEstimation
+        .mul(100 + percentageIncrease)
+        .div(100);
```

Comment:
(Nit, since this is functional as written)

The addition-to-100 followed by divide-by-100 is a bit of a mind-bender.  If we're going to store the percentage as an integer rather than a ratio, something a bit more explicit would be easier to follow

```suggestion
      // Gas estimation is not reliable so we add X% to the stimate to ensure the tx is successful
      const extraGas = gasEstimation.div(percentageIncrease); 
      const adjustedGasLimit = gasEstimation.add(extraGas);
```

Reply:
This is less nitty than it seems. I just copied the way @glitch003 did it in trusting that would work, but there is an edge case that blows this up: `ethers.BigNumber` throw `underflow` when dealing with fractional numbers

So, we cannot do `gasEstimation.mul(1.1)` but we also cannot do `gasEstimation.mul(100+0.1).div(100)` which was an option with `typeof percentageIncrease === number`

I changed it to be directly defined as `BigNumber` in params and the constant at the top. I think that will be safer

Reply:
:+1: rock on man. I like it.